core/server/master/src/main/java/alluxio/master/meta/UpdateCheck.java

```diff
@@ -79,11 +80,22 @@ public static String getLatestVersion(String clusterID, long connectionRequestTi
 
   /**
    * @param clusterID the cluster ID
-   * @return a string representation of the user's environment in the format "key1:value1, key2:
-   *         value2".
+   * @return a string representation of the user's environment in the format
+   *         "Alluxio/{ALLUXIO_VERSION} (valueA; valueB)"
    */
   @VisibleForTesting
   public static String getUserAgentString(String clusterID) throws IOException {
+    Joiner joiner = Joiner.on("; ").skipNulls();
+    String sysInfo = joiner.join(getUserAgentEnvironmentString(clusterID), getFeatureString());
+    return String.format("Alluxio/%s (%s)", ProjectConstants.VERSION, sysInfo);
+  }
+
+  /**
+   * @param clusterID the cluster ID
+   * @return a string representation of the user's environment in the format "docker; kubernetes"
+   */
+  @VisibleForTesting
+  public static String getUserAgentEnvironmentString(String clusterID) throws IOException {
```

Comment:
either this method should be named `getEnvironmentString` or the below method should be `getUserAgentFeatureString` to be consistent with each other